### PR TITLE
Update gradle/gradle android plugin version

### LINKED ISSUE #290: Update Gradle version to 7.5
We currently support Gradle 6.1.1 and Android Gradle plugin 4.0.1

https://github.com/defold/extender/blob/dev/server/docker-base/Dockerfile#L451-L452

Newer minimum versions here: https://developer.android.com/build/releases/gradle-plugin#updating-gradle

### PR DESCRIPTION
**Pull request by ekharkunov**
* Update Gradle to 8.4
* Gradle android plugin to 8.2.1
* Update gradle wrapper that used for building to 8.4. 
* Update build scripts according to new version to Gradle.
* Change gradle docker plugin.
* Use jdk 17 to run gradle during Android build.
* Remove various workarounds with jdk certificates from Dokerfile.

Fixes #290.

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -109,6 +109,7 @@ public class Version {
     def template = engine.createTemplate(text).make(binding)
 
     new File(projectDir.toString() + "/src/main/java/com/defold/extender", "Version.java").text = template.toString()
+    project.ext.set("outSha", sha)
```
Comment (ekharkunov) [inline server/build.gradle:112]: Add commit sha as shared property to use it later during docker image build.

File: server/build.gradle [hunk 2]
```diff
@@ -19,11 +16,14 @@ plugins {
     id 'idea'
     id 'eclipse'
     id 'org.springframework.boot' version '2.6.6'
+    id 'com.bmuschko.docker-remote-api' version '7.2.0'
 }
 
-apply plugin: 'docker'
+apply plugin: 'com.bmuschko.docker-remote-api'
```
Comment (JCash) [inline server/build.gradle:22]: 👍

File: server/build.gradle [hunk 3]
```diff
@@ -117,44 +118,55 @@ compileJava {
...
     }
 }
 
+task buildDocker(type: DockerBuildImage) {
+    dependsOn 'build', ':manifestmergetool:mainJar', 'dockerStageDir'
+    inputDir = dockerBuildDir
+    imageId = bootJar.archiveBaseName
+    dockerFile = new File(dockerBuildDir, 'Dockerfile')
+    images.add("extender/extender:latest")
+    images.add("extender/extender:" + project.outSha)
```
Comment (ekharkunov) [inline server/build.gradle:167]: Produce images with different tags. It helps to identify builds from different commit.

File: server/docker-base/Dockerfile [hunk 4]
```diff
@@ -586,35 +584,5 @@ RUN \
   apt-get remove -y apt-transport-https xvfb && \
   apt-get clean autoclean autoremove
 
-
-# To avoid the mysterious error "Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty"
```
Comment (JCash) [inline server/docker-base/Dockerfile:590]: Oooh, I've been wanting to remove this for sooooo long!

**Review by JCash — APPROVED** [review]

File: server/docker-base/template.build.gradle [hunk 5]
```diff
@@ -24,6 +24,8 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
+    namespace = "com.defold.defold"
```
Comment (ekharkunov) [inline server/docker-base/template.build.gradle:28]: New requirement in latest gradle android plugin. https://developer.android.com/build/configure-app-module#set-namespace

File: server/docker-base/template.gradle.properties [hunk 6]
```diff
@@ -4,4 +4,4 @@ android.enableJetifier={{android-enable-jetifier}}
 # Gradle will stop resolving dependencies if android.useAndroidX is false and a dependency is using Android X
 android.useAndroidX={{android-enable-jetifier}}
 
-org.gradle.java.home=/usr/lib/jvm/java-11-openjdk-amd64
+org.gradle.java.home=/usr/local/jdk-17.0.7+7
```
Comment (ekharkunov) [inline server/docker-base/template.gradle.properties:7]: Use jdk 17 to run Gradle.

**Comment by ekharkunov** [comment]
For testing I commented everything that doesn't relate to Android build because I don't have possibility to build wind sdk, nintendo, ps4 sdks. So if you face any issue - let me know.

**Pull request merged**